technical-reports/color/color-type.md

```diff
@@ -1,22 +1,768 @@
+```
+
+</aside>
+
+While both examples will render as white, the first example is more explicit about the fact that the hue is not applicable. This is important when interpolating between colors or mixing colors, as using each color in a can yield different results.
+
+## Supported Color spaces
+
+The following values are supported for the `colorSpace` property. The `components` array will vary depending on the color space.
+
+<table class="def">
+    <thead>
+        <tr>
+            <th scope="col">Color Space</th>
+            <th scope="col">Key</th>
+            <th scope="col" colspan="2">Values</th>
+        </tr>
+    </thead>
+    <tbody>
+        <!-- sRGB -->
+        <tr>
+            <th scope="rowgroup" rowspan="3">sRGB</th>
+            <td rowspan="3">`"srgb"`</td>
+            <td>Red</td>
+            <td>[0 - 1]</td>
+        </tr>
+        <tr>
+            <td>Green</td>
+            <td>[0 - 1]</td>
+        </tr>
+        <tr>
+            <td>Blue</td>
+            <td>[0 - 1]</td>
+        </tr>
+        <!-- sRGB linear -->
+        <tr>
+            <th scope="rowgroup" rowspan="3">sRGB linear</th>
+            <td rowspan="3">`"srgb-linear"`</td>
+            <td>Red</td>
+            <td>[0 - 1]</td>
+        </tr>
+        <tr>
+            <td>Green</td>
+            <td>[0 - 1]</td>
+        </tr>
+        <tr>
+            <td>Blue</td>
+            <td>[0 - 1]</td>
+        </tr>
+        <!-- HSL -->
+        <tr>
+            <th scope="rowgroup" rowspan="3">HSL</th>
+            <td rowspan="3">`"hsl"`</td>
+            <td>Hue</td>
+            <td>[0 - 360)</td>
+        </tr>
+        <tr>
+            <td>Saturation</td>
+            <td>[0 - 100]</td>
+        </tr>
+        <tr>
+            <td>Lightness</td>
+            <td>[0 - 100]</td>
+        </tr>
+        <!-- HWB -->
+        <tr>
+            <th scope="rowgroup" rowspan="3">HWB</th>
+            <td rowspan="3">`"hwb"`</td>
+            <td>Hue</td>
+            <td>[0 - 360)</td>
+        </tr>
+        <tr>
+            <td>Whiteness</td>
+            <td>[0 - 100]</td>
+        </tr>
+        <tr>
+            <td>Blackness</td>
+            <td>[0 - 100]</td>
+        </tr>
+        <!-- CIELAB -->
+        <tr>
+            <th scope="rowgroup" rowspan="3">CIELAB</th>
+            <td rowspan="3">`"lab"`</td>
+            <td>L</td>
+            <td>[0 - 100]</td>
+        </tr>
+        <tr>
+            <td>A</td>
+            <td>[-Infinity - Infinity]*</td>
+        </tr>
+        <tr>
+            <td>B</td>
+            <td>[-Infinity - Infinity]*</td>
+        </tr>
+        <!-- LCH -->
+        <tr>
+            <th scope="rowgroup" rowspan="3">LCH</th>
+            <td rowspan="3">`"lch"`</td>
+            <td>L</td>
+            <td>[0 - 100]</td>
+        </tr>
+        <tr>
+            <td>C</td>
+            <td>[0 - Infinity]**</td>
+        </tr>
+        <tr>
+            <td>Hue</td>
+            <td>[0 - 360)</td>
```

Comment:
What does the parenthesis mean here?

Reply:
[] is inclusive, () is exclusive. ie, if a range is (0-1), everything _between_ 0 and 1 is in the range, but 0 and 1 are not. 

In the case of polar coordinates, 0 and 360 degrees are equivalent. So, the minimum range to represent every possible hue angle is [0-360). If we allow 360, transformers will be expected to re-normalize the range; so I figure we can eliminate some complexity by excluding 360.

The CSS Color Module 4 spec says this:

> This number is normalized to the range [0,360)[^1]

Do you think we need to explain this in our spec?

[^1]: https://www.w3.org/TR/css-color-4/#hue-syntax

Reply:
I went ahead and added an explanation to the top of the table :)